pyoptsparse/pyOpt_optimizer.py

```diff
@@ -798,9 +805,9 @@ def setOption(self, name, value=None):
             Variable value to set.
         """
 
-        # Call the optimizer specific routine
+        super().setOption(name, value)
```

Comment:
I was wondering why this was previously set after `_on_setOption`
To clarify, `super().setOption()` sets the options in the python layer via the inherited `BaseSolver` method, while `_on_setOption` is the actual optimizer routine that (when available, eg. for SNOPT and IPOPT) passes the options from the python layer to the optimizer "binary", right?

Reply:
Not exactly. `super().setOption()` calls the function in BaseSolver, which just sets the options dictionary. Several optimizers maintained a separate `set_options` variable which stores the options which are set by the user rather than by default. These are then passed to the optimizers at a later place (usually during `__call__` since it requires an optimizer instance).

Reply:
Ok yeah I got the inherited method and I now I follow how the info is passed, I did not realize that the child `pySNOPT` instance overwrites `_on_setOption` in the inherited `setOption()`, sometimes I miss these OOP tricks

Reply:
Yeah it is overloaded in the child classes. Very few optimizers use those but they are there.